### ISSUE TITLE
Use the name "frontmatter" for JSX data exports

### ISSUE THREAD

**Issue by kripod** [gatsbyjs/gatsby#3795]
[Currently](https://github.com/gatsbyjs/gatsby/blob/dcfa0fab32e168c075c052dde7ecd97736c6147a/packages/gatsby-transformer-javascript-static-exports/README.md), `exports.data = { /* ... */ }` is being used, which could also be written as `export const data = { /* ... */ }`, but the latter option can be inconvenient _(variable name clashes may occur)_ when a GraphQL query is also in place for the given JSX.

I would like to propose using `export const frontmatter = { /* ... */ }` in order to avoid confusion and improve consistence with markdown file handling.

**Comment by jbolda** [comment]
In the past, I had more plans for this so I went with the more generic data. Those plans are pretty much now defunct though, and it seems to be only worthwhile to use this for frontmatter. I think, in an effort of discoverability and backward compatibility, we actually deprecate this package. Then we make a new package, `gatsby-transformer-javascript-frontmatter`, and make your proposed change. I think this will help others discover this package as I have noticed in the past that this package isn't quite obvious what it does at first glance. @KyleAMathews, are you amenable to this?
